src/cascadia/TerminalControl/XamlUiaTextRange.cpp

```diff
@@ -89,12 +90,56 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     winrt::Windows::Foundation::IInspectable XamlUiaTextRange::GetAttributeValue(int32_t textAttributeId) const
     {
-        // Copied functionality from Types::UiaTextRange.cpp
-        if (textAttributeId == UIA_IsReadOnlyAttributeId)
+        // Call the function off of the underlying UiaTextRange.
+        VARIANT result;
+        THROW_IF_FAILED(_uiaProvider->GetAttributeValue(textAttributeId, &result));
+
+        // Convert the resulting VARIANT into a format that is consumable by XAML.
+        switch (result.vt)
+        {
+        case VT_BSTR:
+        {
+            return box_value(result.bstrVal);
+        }
+        case VT_I4:
+        {
+            return box_value(result.iVal);
+        }
+        case VT_R8:
+        {
+            return box_value(result.dblVal);
+        }
+        case VT_BOOL:
+        {
+            return box_value(result.boolVal);
+        }
+        case VT_UNKNOWN:
         {
-            return winrt::box_value(false);
+            // This one is particularly special.
+            // We might return a special value like UiaGetReservedMixedAttributeValue
+            //  or UiaGetReservedNotSupportedValue.
+            // Some text attributes may return a real value, however, none of those
+            //  are supported at this time.
+            // So we need to figure out what was actually intended to be returned.
+
+            IUnknown* notSupportedVal;
+            UiaGetReservedNotSupportedValue(&notSupportedVal);

```

Comment:
*hold the phone*

I think that UiaGetReservedXxx _returns an owning reference_, which you must store in a `winrt::com_ptr` or `WRL::ComPtr` (depending on the style in use in this file)